bin/docker/Dockerfile

```diff
@@ -0,0 +1,43 @@
+# Not alpine, due to possible performance issues of MUSL malloc.
+#
+# In theory this should not be relevant because PHP uses its own allocator,
+# but some one-time initialization logic inside PHP bypasses it,
+# which means system malloc *is* used more often especially in cases like these.
+
+FROM php:8.4-cli
+
+# This line invalidates cache when master branch changes
+ADD https://github.com/vimeo/psalm/commits/master.atom /dev/null
+
+ADD https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
+
+RUN sed 's/-O2/-O3/g' -i /usr/local/bin/install-php-extensions && \
+    chmod +x /usr/local/bin/install-php-extensions && \
+    install-php-extensions pcntl mbstring xml dom igbinary opcache && \
+    rm /usr/local/bin/install-php-extensions
+
+RUN apt-get update && apt-get -y --no-install-recommends install git unzip && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+ADD bin/docker/php.ini /usr/local/lib/php.ini
+
+COPY --from=composer:2 /usr/bin/composer /usr/bin/composer
+
+ARG PSALM_REV=dev-master
+RUN COMPOSER_ALLOW_SUPERUSER=1 \
+    COMPOSER_HOME="/composer" \
+    composer global require vimeo/psalm:${PSALM_REV} --prefer-dist --no-progress --dev && \
```

Comment:
Won't this suffer from a race condition?

E.g. when you're building a container for a new, freshly tagged release you'd want packagist.org to pick it up first, otherwise this would fail, wouldn't it?

Reply:
That is indeed correct, and I thought of this already, will add a waiting loop later